pkg/clusteragent/autoscaling/workload/model/pod_autoscaler.go

```diff
@@ -618,3 +671,13 @@ func getLongestScalingRulesPeriod(rules []datadoghq.DatadogPodAutoscalerScalingR
 
 	return longest
 }
+
+func parseCustomConfigurationAnnotation(annotations map[string]string) *RecommenderConfiguration {
+	customConfiguration := RecommenderConfiguration{}
+
+	if err := json.Unmarshal([]byte(annotations[AnnotationsConfigurationKey]), &customConfiguration); err != nil {
+		log.Debugf("Failed to parse annotations for custom recommender configuration: %v", err)
```

Comment:
We should probably record this in error and show it in the status of DPA

Reply:
In the case that no annotation is set, this would also return an error. Should it be changed to
1. Check if the annotation key exists
2. Try to unmarshal (and update `p.error` here if it cannot)

Reply:
updated!